tests/unit-tests/frontend_wayland/test_desktop_file_manager.cpp

```diff
@@ -0,0 +1,282 @@
+    auto new_file = std::make_shared<mf::DesktopFile>(gnome_prefixed_desktop_file_app_id.c_str(), nullptr, nullptr);
+    cache->files.push_back(new_file);
+    auto app_id = file_manager->resolve_app_id(&surface);
+    EXPECT_THAT(app_id, Eq(new_file->id));
+}
+
+struct DesktopFileManagerParameterizedTestFixture : public DesktopFileManager, WithParamInterface<std::string> {
+};
+
+TEST_P(DesktopFileManagerParameterizedTestFixture, can_find_app_when_app_id_matches_with_prefix)
+{
+    std::string prefix = GetParam();
+    const std::string prefixed = prefix + DESKTOP_FILE_APP_ID;
+    auto new_file = std::make_shared<mf::DesktopFile>(prefixed.c_str(), nullptr, nullptr);
+    cache->files.push_back(new_file);
+    auto app_id = file_manager->resolve_app_id(&surface);
+    EXPECT_THAT(app_id, Eq(new_file->id));
+}
+
+INSTANTIATE_TEST_SUITE_P(DesktopFileManager, DesktopFileManagerParameterizedTestFixture, ::testing::Values(
+    "gnome-",
+    "fedora-",
+    "mozilla-",
+    "debian-"
+));
+
+TEST_F(DesktopFileManager, when_security_profile_does_not_start_with_prefix_then_empty_string_is_returned)
+{
+    auto result = mf::DesktopFileManager::parse_snap_security_profile_to_desktop_id("firefox.firefox (current");
+    EXPECT_TRUE(result.empty());
+}
+
+TEST_F(DesktopFileManager, when_security_profile_is_valid_then_desktop_id_is_returned)
+{
+    auto result = mf::DesktopFileManager::parse_snap_security_profile_to_desktop_id("snap.firefox.firefox (enforce)");
+    EXPECT_EQ(result, "firefox_firefox.desktop");
+}
+
+TEST_F(DesktopFileManager, when_security_profile_is_valid_and_lacks_protection_indication_then_desktop_id_is_returned)
+{
+    auto result = mf::DesktopFileManager::parse_snap_security_profile_to_desktop_id("snap.firefox.firefox");
+    EXPECT_EQ(result, "firefox_firefox.desktop");
+}
+
+TEST_F(DesktopFileManager, can_find_desktop_file_by_exactly_matching_exec_name)
+{
+    auto const app_exec = "gnome-calendar";
+    auto new_file = std::make_shared<mf::DesktopFile>(nullptr, nullptr, app_exec);
+    cache->files.push_back(new_file);
+    auto desktop_file = file_manager->parse_exec_call_to_desktop_file(app_exec);
+    EXPECT_THAT(desktop_file, Eq(new_file));
+}
+
+TEST_F(DesktopFileManager, can_find_desktop_file_if_file_in_exec_path_matches_name)
+{
+    auto new_file = std::make_shared<mf::DesktopFile>(nullptr, nullptr, "gnome-calendar");
+    cache->files.push_back(new_file);
+    auto desktop_file = file_manager->parse_exec_call_to_desktop_file("/usr/bin/gnome-calendar");
+    EXPECT_THAT(desktop_file, Eq(new_file));
+}
+
+TEST_F(DesktopFileManager, can_find_desktop_file_if_exec_name_matches_despite_arguments_given)
+{
+    auto new_file = std::make_shared<mf::DesktopFile>(nullptr, nullptr, "gnome-calendar");
+    cache->files.push_back(new_file);
+    auto desktop_file = file_manager->parse_exec_call_to_desktop_file("gnome-calendar --arg1 --arg2");
+    EXPECT_THAT(desktop_file, Eq(new_file));
+}
```

Comment:
Pretty sure this is unnecessary, because your input data doesn't look like "gnome-calendar --arg1 --arg2".